extensions/32-bit-address-space/README.md

```diff
@@ -0,0 +1,58 @@
+# General Design
+
+**Extension State: Under Development**
+**Requires: CP1.14**
+**CP Bit: CP1.16**
+
+# Overview
+
+This extension adds a new concept of _modes_ to the ISA. Modes are used to control things about the processor
+state which are binary-incompatible - that is, binary code written for a mode other than the mode the
+processor is actually in will almost certainly behave incorrectly.
+
+In particular, this extension focuses on the _Real 32-bit address mode_.
+
+# Added Instructions
+
+None!
+
+# Modes
+
+CR `1110`, equivalently `cr14` or "the MODE control register," holds a value indicating the current
+_mode_ of the processor.
+
+## Bare Mode
+
+The mode described by [the base isa](../../base-isa.md) is known as the
+Bare mode (Real mode?). In that mode, pointers are 16 bits and there is no virtual memory, paging,
+memory protection, safe multitasking, or privilege restriction. Bare mode is indicated by a value of
+0 in `cr14`.
```

Comment:
Maybe this should be called base mode or initial mode? Also, privilege restriction has it's own CR so privilege restrictions can still occur without this. Does safe multitasking need to be explicitly mentioned? Isn't that a combined property of interrupts + memory protection?

Reply:
"Base" mode seems like a good name. I'll update that.
I think in terms of "safe multitasking" I was thinking more like multicore cache ownership and stuff like that. But yeah, as long as the system supports caching in real mode at all, it seems orthogonal to this, so I'll remove that bit. Simply no virtual memory, paging, or memory protection (as memory protection is typically part of paging).